src/components/FadedScrollCard.tsx

```diff
@@ -0,0 +1,281 @@
+  initialScrollEnabled?: boolean;
+  isExpanded: boolean;
+  onPressCollapsedCard?: () => void;
+  skipCollapsedState?: boolean;
+};
+
+export const FadedScrollCard = ({
+  cardHeight,
+  children,
+  contentHeight,
+  expandedCardBottomInset = 120,
+  expandedCardTopInset = 120,
+  initialScrollEnabled,
+  isExpanded,
+  onPressCollapsedCard,
+  skipCollapsedState,
+}: FadedScrollCardProps) => {
+  const { height: deviceHeight, width: deviceWidth } = useDimensions();
+  const { isDarkMode } = useTheme();
+
+  const cardRef = useAnimatedRef<Animated.View>();
+
+  const [scrollEnabled, setScrollEnabled] = useState(initialScrollEnabled);
+  const [isFullyExpanded, setIsFullyExpanded] = useState(false);
+
+  const yPosition = useSharedValue(0);
+
+  const maxExpandedHeight = deviceHeight - (expandedCardBottomInset + expandedCardTopInset);
+
+  const containerStyle = useAnimatedStyle(() => {
+    return {
+      height:
+        cardHeight.value > MAX_CARD_HEIGHT || !skipCollapsedState
+          ? interpolate(
+              cardHeight.value,
+              [MAX_CARD_HEIGHT, MAX_CARD_HEIGHT, maxExpandedHeight],
+              [cardHeight.value, MAX_CARD_HEIGHT, MAX_CARD_HEIGHT],
+              'clamp'
+            )
+          : undefined,
+      zIndex: interpolate(cardHeight.value, [0, MAX_CARD_HEIGHT, MAX_CARD_HEIGHT + 1], [1, 1, 2], 'clamp'),
+    };
+  });
+
+  const backdropStyle = useAnimatedStyle(() => {
+    const canExpandFully = contentHeight.value + CARD_BORDER_WIDTH * 2 > MAX_CARD_HEIGHT;
+    return {
+      opacity: canExpandFully && isFullyExpanded ? withTiming(1, timingConfig) : withTiming(0, timingConfig),
+    };
+  });
+
+  const cardStyle = useAnimatedStyle(() => {
+    const canExpandFully = contentHeight.value + CARD_BORDER_WIDTH * 2 > MAX_CARD_HEIGHT;
+    const expandedCardHeight = Math.min(contentHeight.value + CARD_BORDER_WIDTH * 2, maxExpandedHeight);
+    return {
+      borderColor: interpolateColor(
+        cardHeight.value,
+        [0, MAX_CARD_HEIGHT, expandedCardHeight],
+        isDarkMode ? ['#1F2023', '#1F2023', '#242527'] : ['#F5F7F8', '#F5F7F8', '#FBFCFD']
+      ),
+      height: cardHeight.value > MAX_CARD_HEIGHT ? cardHeight.value : undefined,
+      position: canExpandFully && isFullyExpanded ? 'absolute' : 'relative',
+      transform: [
+        {
+          translateY: interpolate(
+            cardHeight.value,
+            [0, MAX_CARD_HEIGHT, expandedCardHeight],
+            [
+              0,
+              0,
+              -yPosition.value +
+                expandedCardTopInset +
+                (deviceHeight - (expandedCardBottomInset + expandedCardTopInset) - expandedCardHeight) -
+                (yPosition.value + expandedCardHeight >= deviceHeight - expandedCardBottomInset
+                  ? 0
+                  : deviceHeight - expandedCardBottomInset - yPosition.value - expandedCardHeight),
+            ]
+          ),
+        },
+      ],
+    };
+  });
```

Comment:
This is kind of difficult to read. Could we use a function to grab these values so that we can break up these conditionals into named values so it's a little bit easier to read and reason about?